### PR TITLE
feat(environment): Add name

### PR DESCRIPTION
**Pull request by gagantrivedi**

### CODE REVIEW

File: tests/unit/environments/test_environments_builders.py [hunk 1]
```diff
@@ -54,6 +54,38 @@ def test_build_environment_model():
...
+        "feature_states": [],
+    }
+
+    # When
+    environment_model = build_environment_model(environment_dict)
+
+    # Then
+    assert environment_model.name == environment_name
+
+    # assert len(environment_model.feature_states) == 1
```
Comment (matthewelwell) [inline tests/unit/environments/test_environments_builders.py:85]: Why are these commented out?

Comment (gagantrivedi) [reply tests/unit/environments/test_environments_builders.py]: my  bad forgot to remove it

**Review by matthewelwell — APPROVED** [review]

File: flag_engine/environments/schemas.py [hunk 2]
```diff
@@ -38,8 +38,10 @@ class Meta:
 class BaseEnvironmentSchema(Schema):
     id = fields.Int()
     api_key = fields.Str()
-    allow_client_traits = fields.Bool(required=False, default=True)
     updated_at = fields.DateTime()
+
+    name = fields.Str(requried=False)
```
Comment (matthewelwell) [inline flag_engine/environments/schemas.py:43]: Ah, hang on. Typo!

**Review by matthewelwell — CHANGES_REQUESTED** [review]

File: flag_engine/environments/schemas.py (hunk 2 above)
Comment (gagantrivedi) [reply flag_engine/environments/schemas.py:43]: ah

**Review by matthewelwell — APPROVED** [review]

**Pull request merged**